server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -314,18 +290,35 @@ protected BytesRef indexedValueForSearch(Object value) {
         }
     }
 
-    private int ignoreAbove;
-    private boolean splitQueriesOnWhitespace;
-    private String nullValue;
+    private final boolean indexed;
+    private final boolean hasDocValues;
+    private final String nullValue;
+    private final boolean eagerGlobalOrdinals;
+    private final int ignoreAbove;
+    private final String indexOptions;
+    private final FieldType fieldType;
+    private final SimilarityProvider similarity;
+    private final String normalizerName;
+    private final boolean splitQueriesOnWhitespace;
+
+    private final IndexAnalyzers indexAnalyzers;
 
     protected KeywordFieldMapper(String simpleName, FieldType fieldType, MappedFieldType mappedFieldType,
-                                 int ignoreAbove, boolean splitQueriesOnWhitespace, String nullValue,
-                                 MultiFields multiFields, CopyTo copyTo) {
-        super(simpleName, fieldType, mappedFieldType, multiFields, copyTo);
+                                 MultiFields multiFields, CopyTo copyTo, Builder builder) {
+        super(simpleName, mappedFieldType, multiFields, copyTo);
         assert fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS) <= 0;
-        this.ignoreAbove = ignoreAbove;
-        this.splitQueriesOnWhitespace = splitQueriesOnWhitespace;
-        this.nullValue = nullValue;
+        this.indexed = builder.indexed.getValue();
```

Comment:
Are we planning to use all these parameters later instead of accessing them through fieldType() as below:
e.g.  =>
fieldType().hasDocValues() => hasDocValues?

Do we still need a `fieldType` here, if all values can be retrieved through `builder`?

Reply:
We still need `fieldType` because it needs to be passed to the super constructor, although I would like to make it unnecessary in future.  The fields themselves are used in the Parameter definitions further up, to initialize the merge builder from the existing mapping.